src/server/src/routes/document/index.ts

```diff
@@ -0,0 +1,139 @@
+import StatusCodes from 'http-status-codes';
+import { Request, Response, Router } from 'express';
+import { Document } from '@entities/document';
+import { errors } from '@shared/errors';
+import {
+    createDocument,
+    getDocuments,
+    deleteDocument,
+} from '@modules/document';
+import { JWTUser } from '@entities/user';
+import logger from '@shared/Logger';
+import passport from 'passport';
+
+const router = Router();
+const {
+    CREATED,
+    OK,
+    INTERNAL_SERVER_ERROR,
+    UNAUTHORIZED,
+    BAD_REQUEST,
+} = StatusCodes;
+
+interface docRequest extends Request {
+    body: {
+        document: Document;
+    };
+}
+
+/******************************************************************************
+ *            POST Request - Create - /api/document/create
+ ******************************************************************************/
+
+ router.post(
+     '/create',
+     passport.authenticate('jwt', { session: false }),
+     async (req: docRequest, res: Response) => {
+        const { role, specificUserId } = req.user as JWTUser;
+        if (role !== 'student') {
+            return res
+                .status(UNAUTHORIZED)
+                .json({ error: 'User is not a student' })
+                .end();
+        }
+
+        const {
+            name,
+            docType,
+            isDefault,
+            //dateAdded,
+            document,
+        } = req.body.document;
+        try {
+            let document1 = Buffer.alloc(16);
+            const { result, message } = await createDocument(
+                name,
+                docType,
+                isDefault,
+                //dateAdded,
+                document1,
```

Comment:
Add a comment like. // TODO: In the future replace document1 with document data received in the request